### PR TITLE
std.meta: add isError

### PR DESCRIPTION
**Pull request by xackus**

### CODE REVIEW

**Comment by LewisGaul** [comment]
Thanks for adding this, looks good to me (not sure whether I should be offering a formal review).

File: lib/std/meta.zig [hunk 1]
```diff
@@ -1334,3 +1334,13 @@ test "shuffleVectorIndex" {
     testing.expect(shuffleVectorIndex(6, vector_len) == -3);
     testing.expect(shuffleVectorIndex(7, vector_len) == -4);
 }
+
+/// Returns whether `error_union` contains an error.
+pub fn isError(error_union: anytype) bool {
+    return if(error_union) |_| false else |_| true;
```
Comment (LewisGaul) [inline lib/std/meta.zig:1340]: Doesn't zig fmt add a space after the 'if'? I thought the stdlib was run through zig fmt?

Comment (g-w1) [reply lib/std/meta.zig:1340]: That is blocking on #8219.

**Comment by ehaas** [comment]
Would it make sense to expand `isError` to work for all types? Right now it's a compile error if `error_union` isn't an error union, but there could be comptime checks for `@typeInfo(@TypeOf(param)) == .ErrorSet` and `@typeInfo(@TypeOf(param)) == .ErrorUnion`

File: lib/std/json/test.zig [hunk 2]
```diff
@@ -29,8 +29,7 @@ fn ok(s: []const u8) !void {
 fn err(s: []const u8) void {
     testing.expect(!json.validate(s));
 
-    testNonStreaming(s) catch return;
-    testing.expect(false);
+    testing.expect(std.meta.isError(testNonStreaming(s)));
```
Comment (andrewrk) [inline lib/std/json/test.zig:32]:
```suggestion
    testing.expectError(error.WhateverTheErrorIs, testNonStreaming(s));
```

**Review by andrewrk — CHANGES_REQUESTED** [review]
Implementation looks fine but existing tests should be updated to use `testing.expectError` rather than this function.

File: lib/std/json/test.zig (hunk 2 above)
Comment (andrewrk) [reply lib/std/json/test.zig:32]: on second thought this is a pretty weird looking function and your code is definitely an improvement. happy to merge as is :+1:

**Pull request merged**